### PR TITLE
[MM-20469] Transformed emoji_list_item to ts

### PR DESCRIPTION
**Pull request by Ayanrocks**
Migrated components/emoji/emoji_list_item' module and associated tests to TypeScript

#### Ticket Link
Fixes: https://github.com/mattermost/mattermost-server/issues/15471
JIRA:  https://mattermost.atlassian.net/browse/MM-20469

### CODE REVIEW

**Comment by Ayanrocks** [comment]
@ashishbhate Review

**Review by ashishbhate — APPROVED** [review]

File: components/emoji/emoji_list_item/emoji_list_item.tsx [hunk 1]
```diff
@@ -2,43 +2,42 @@
...
-        creatorDisplayName: PropTypes.string.isRequired,
-        creatorUsername: PropTypes.string,
-        currentTeam: PropTypes.object,
-        onDelete: PropTypes.func,
-        actions: PropTypes.shape({
-            deleteCustomEmoji: PropTypes.func.isRequired,
-        }).isRequired,
+export type Props = {
+    emoji: CustomEmoji;
+    emojiId?: string;
```
Comment (cpoile) [inline components/emoji/emoji_list_item/emoji_list_item.tsx:15]: I don't think this needs to be in the props here.

**Review by cpoile — CHANGES_REQUESTED** [review]
Looks good, thanks @Ayanrocks! One little question below.

File: components/emoji/emoji_list_item/emoji_list_item.tsx (hunk 1 above)
Comment (Ayanrocks) [reply components/emoji/emoji_list_item/emoji_list_item.tsx:15]: It is used in index.ts and read this PR thread https://github.com/mattermost/mattermost-webapp/pull/6742

Comment (cpoile) [reply components/emoji/emoji_list_item/emoji_list_item.tsx:15]: Ahh, so it is. Thanks @Ayanrocks !

**Review by cpoile — APPROVED** [review]
Thanks!

**Comment by Ayanrocks** [comment]
is there a reason why it's not being merged? Did I did something wrong? 😕 @cpoile @ashishbhate

**Comment by ashishbhate** [comment]
@Ayanrocks apologies for the delay. Its the holiday season so reviews are happening a bit slow. Absolutely nothing wrong on your end :)

**Comment by ashishbhate** [comment]
@Ayanrocks - There are some minor lint issues:

```
mattermost-webapp/components/emoji/emoji_list_item/emoji_list_item.tsx
  16:26  error  Expected a semicolon  @typescript-eslint/member-delimiter-style
  19:41  error  Expected a semicolon  @typescript-eslint/member-delimiter-style
  21:59  error  Expected a semicolon  @typescript-eslint/member-delimiter-style
  22:6   error  Expected a semicolon  @typescript-eslint/member-delimiter-style
```

**Comment by Ayanrocks** [comment]
This can be merge now @ashishbhate

**Comment by ashishbhate** [comment]
Thanks for the effort @Ayanrocks! We can merge this as soon as QA approves the PR. Things are a bit slow at the moment because of the year end break.

**Review by lindalumitchell — APPROVED** [review]
Looks good to me, thanks!

**Pull request merged**